tools/dockerfile/push_testing_images.sh

```diff
@@ -67,6 +67,22 @@ ALL_DOCKERFILE_DIRS=(
 
 CHECK_FAILED=""
 
+if [ "${CHECK_MODE}" != "" ]
+then
+  # Check that there are no stale .current_version files (for which the corresponding
+  # dockerfile_dir doesn't exist anymore).
+  for CURRENTVERSION_FILE in $(find tools/ third_party/rake-compiler-dock -name '*.current_version')
+  do
+    DOCKERFILE_DIR="$(echo ${CURRENTVERSION_FILE} | sed 's/.current_version$//')"
+    if [ ! -e "${DOCKERFILE_DIR}/Dockerfile" ]
+    then
+       echo "Found that ${DOCKERFILE_DIR} has '.current_version' file but there is no corresponding Dockerfile."
+       echo "Should the ${CURRENTVERSION_FILE} file be deleted?"
+       CHECK_FAILED=true
```

Comment:
Why not `read` a prompt and `rm` here? You could also collect the set of unmatched current_version files and do a single read/rm prompt afterwards, I'm not sure when I'd want to keep them.

Reply:
Actually, I'm not sure how you'd detect if it's running interactively. You can disregard this idea.